### PR TITLE
Expose proof base64 conversion

### PR DESCRIPTION
**Pull request by mitschabaude**
low-level method exposed to have more flexibility for how to serialize proofs

I need this in https://github.com/zksecurity/mina-credentials/pull/82 to get a synchronous version of `Proof.fromJSON()`

### CODE REVIEW

**Review by Geometer1729 — APPROVED** [review]
LGTM

**Comment by mitschabaude** [comment]
that changelog CI job is effective :D

File: src/lib/proof-system/proof.ts [hunk 1]
```diff
@@ -92,6 +92,13 @@ class ProofBase<Input = any, Output = any> {
   publicFields() {
     return (this.constructor as typeof ProofBase).publicFields(this);
   }
+
+  static _proofFromBase64(proofString: string, maxProofsVerified: 0 | 1 | 2) {
```
Comment (45930) [inline src/lib/proof-system/proof.ts:96]: In the existing `fromJSON` implementation, we await `initializeBindings`.  What are the implications of skipping that step before entering pickles?

**Review by Shigoto-dev19 — COMMENTED** [review]
looks good!

File: src/lib/proof-system/proof.ts [hunk 2]
```diff
@@ -92,6 +92,13 @@ class ProofBase<Input = any, Output = any> {
...
+    return Pickles.proofOfBase64(proofString, maxProofsVerified)[1];
```
Comment (45930) [inline src/lib/proof-system/proof.ts:97]:
```suggestion
    if (!areBindingsInitialized) {
      throw Error(
        'Pickles bindings must be initialized before calling this method'
      );
    }
    return Pickles.proofOfBase64(proofString, maxProofsVerified)[1];
```

What if we did something like this, coupled with update the export in snarky.js

```
export {
  Snarky,
  Ledger,
  Pickles,
  Test,
  withThreadPool,
  wasm,
  initializeBindings,
  isInitialized as areBindingsInitialized, <-- add this
};
```

That way no one accidentally gets into trouble with this.

**Comment by mitschabaude** [comment]
@Trivo25 I think this is ready for code owner approval

**Review by Shigoto-dev19 — APPROVED** [review]

**Review by Trivo25 — APPROVED** [review]

**Pull request merged**